src/main/java/macrobase/analysis/transform/BatchScoreFeatureTransform.java

```diff
@@ -13,23 +13,35 @@
     protected BatchTrainScore batchTrainScore;
     protected MacroBaseConf conf;
 
+    protected boolean requiresTraining;
```

Comment:
BatchTrainScore should be invoked for every context. For some contexts, I may want to skip the training because they may have the same distribution as a parent context. I need a switch to control whether to invoke train() or not. 

If we have train() act as a no-op by default, don't we also need a switch anyways to control wether to invoke the actual train()?